src/services/protobufService.ts

```diff
@@ -115,58 +81,89 @@ class ProtobufService {
-        if( err instanceof  Response) {
-          console.error('Schema ' + name + '. Error ' + err.status + ' ' + err.statusText);
-        }
-        return left(<ActionResponse>{ message: errorMessage, success: false });
-      });
+      .then((schema) => right(schema))
+      .catch((err) => this.handleProtobufRestError(err).then((r) => left(r)));
   }
 
   /**
    * Get the list of files and validation response
    */
   public getProtobufSchemas(): Promise<Either<ActionResponse, ProtoSchema[]>> {
+    // @ts-ignore
     return utils
       .restCall(this.endpoint, 'GET')
-      .then(response => {
+      .then((response) => {
         if (response.ok) {
           return response.json();
         }
         throw response;
       })
-      .then(protoSchemas =>
+      .then((protoSchemas) =>
         right(
-          protoSchemas.map(schema => {
-            let protoError: ProtoError | undefined = undefined;
-            if (schema['error'] != null) {
-              protoError = <ProtoError>{
-                message: schema.error.message,
-                cause: schema.error.cause
+          protoSchemas
+            .map((schema) => {
+              let protoError: ProtoError | undefined = undefined;
+              if (schema['error'] != null) {
+                protoError = <ProtoError>{
+                  message: schema.error.message,
+                  cause: schema.error.cause,
+                };
+              }
+              return <ProtoSchema>{
+                name: schema.name,
+                error: protoError,
               };
-            }
-            return <ProtoSchema>{
-              name: schema.name,
-              error: protoError
-            };
-          }).filter(schema => schema.name != 'org/infinispan/protostream/message-wrapping.proto' &&
-            schema.name != 'org/infinispan/query/remote/client/query.proto')
+            })
+            .filter(
+              (schema) =>
+                schema.name !=
+                  'org/infinispan/protostream/message-wrapping.proto' &&
+                schema.name != 'org/infinispan/query/remote/client/query.proto'
+            )
         )
       )
-      .catch(err => left(<ActionResponse>{ message: err, success: false }));
+      .catch((err) => this.handleProtobufRestError(err).then((r) => left(r)));
+  }
+
+  private handleProtobufRestError(err): Promise<ActionResponse> {
+    let actionResponse = <ActionResponse>{
+      message: 'Unknown error retrieving protobuf schemas',
+      success: false,
+    };
+
+    if (err instanceof TypeError) {
+      actionResponse = <ActionResponse>{
+        message: err.message,
+        success: false,
+      };
+    } else if (err instanceof Response) {
+      if (err.status.valueOf() == 403) {
+        // Not Found
+        actionResponse = <ActionResponse>{
+          message:
+            'Unauthorized access. The user lacks ___schema_manager permission.',
```

Comment:
```suggestion
            'Unauthorized access.',
```
I don't think it makes sense to leak the permission required. The permission's name is going to change and, using other sites as an example, they just say "unauthorized".